tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -628,73 +670,124 @@ def _enter_Module(self, node: chapel.Module):
+                    decl_file = node.location().path()
+                    is_standard_decl = self.context.context.is_bundled_path(
+                        decl_file
+                    )
+                    show = is_standard_decl
 
-            file = ast.location().path()
-            in_bundled_module = self.context.context.is_bundled_path(file)
+                if show:
+                    documented_nodes.append(node)
 
-            for name, nodes in scope.visible_nodes():
-                # Don't show internal symbols to the user, even if they
-                # are technically in scope. The exception is if we're currently
-                # editing a standard file.
-                skip_prefixes = ["chpl_", "chpldev_", "_"]
-                if any(name.startswith(prefix) for prefix in skip_prefixes):
-                    if not in_bundled_module:
-                        continue
+            if len(documented_nodes) == 0:
+                return None
 
-                # Only show nodes without @chpldoc.nodoc. The exception
-                # about standard files applies here too.
-                documented_nodes = []
-                for node in nodes:
-                    # apply aforementioned exception
-                    if in_bundled_module:
-                        documented_nodes.append(node)
-                        continue
+            # Just take the first value to avoid showing N entries for
+            # overloaded functions.
+            return name, documented_nodes[0]
 
-                    # avoid nodes with nodoc attribute.
-                    ag = node.attribute_group()
-                    show = False
-                    if not ag or not ag.get_attribute_named("chpldoc.nodoc"):
-                        show = True
-                    elif name in _ALLOWED_NODOC_DECLS:
-                        # If users declare variables like 'here' themselves,
-                        # we will not show them if they're @chpldoc.nodoc,
-                        # since they're not special.
-                        decl_file = node.location().path()
-                        is_standard_decl = self.context.context.is_bundled_path(
-                            decl_file
+        def visible_nodes_for_scopes(
+            node: chapel.AstNode, scopes: List[chapel.Scope]
+        ):
+            visible_nodes = []
+            file = node.location().path()
+            in_bundled_module = self.context.context.is_bundled_path(file)
+            for depth, scope in enumerate(scopes):
+                files_named_in_use_or_import = set()
+                for m in scope.modules_named_in_use_or_import():
+                    files_named_in_use_or_import.add(m.location().path())
+
+                for name, nodes in scope.visible_nodes():
+                    visible_node = visible_nodes_for_scope(
+                        name, nodes, in_bundled_module
+                    )
+                    if visible_node:
```

Comment:
You can also flatten this logic by doing if-false-continue instead of creating increasingly nested ifs.